File: src/main/java/com/team04/back/domain/weather/geo/dto/GeoLocationDto.java
package com.team04.back.domain.weather.geo.dto;

import com.team04.back.infra.weather.dto.GeoDirectResponse;
import jakarta.validation.constraints.NotNull;

public record GeoLocationDto(
        @NotNull String name,
        @NotNull String country,
        @NotNull double lat,
        @NotNull double lon,
        String localName
) {
    public GeoLocationDto(GeoDirectResponse geoDirectResponse) {
        this(
                geoDirectResponse.getName(),
                geoDirectResponse.getCountry(),
                geoDirectResponse.getLat(),
                geoDirectResponse.getLon(),
                geoDirectResponse.getLocalNames() != null
                        ? geoDirectResponse.getLocalNames().getKorean()
                        : null
        );
    }
}

File: src/main/java/com/team04/back/domain/weather/geo/dto/GeoLocationDto.kt
package com.team04.back.domain.weather.geo.dto

import com.team04.back.infra.weather.dto.GeoDirectResponse

data class GeoLocationDto(
    val name: String? = null,
    val country: String? = null,
    val lat: Double = 0.0,
    val lon: Double = 0.0,
    val localName: String? = null
) {
    constructor(geoDirectResponse: GeoDirectResponse) : this(
        name = geoDirectResponse.name,
        country = geoDirectResponse.country,
        lat = geoDirectResponse.lat,
        lon = geoDirectResponse.lon,
        localName = geoDirectResponse.localNames?.korean
    )
}

